resource.go

```diff
@@ -38,8 +38,14 @@ type Condition struct {
 	Value     string  // Value or regular expression to search for
 }
 
+// Output is the response payload returned from an Infoblox instance.
```

Comment:
So responses are no longer arrays from Infoblox, they're objects with a "result" field?  That must be something tied to the new WAPI version.  Again, I want to avoid changing the public API since many users of this library aren't pinned to a version.

Reply:
The "next_page_id" field is returned whenever a page request is send, meaning the "_paging=1", "_return_as_object=1" and "_max_results=(page-size)" query parameters are present in the request URL. If a non-paging request is send, the objects are returned as an array. Here are some examples:

### Initial Page Request
`curl -H "Content-Type: application/json" -k1 -u admin -X GET  'https://ib-prod.example.com/wapi/v2.5/network?_paging=1&_return_as_object=1&_max_results=2&_proxy_search=GM'`

### Initial Page Response
```json  
 "next_page_id": "789c5590d16ec3200c45dffd23c9cba292b66bfb0979a9aaf503100112b1251881a3b67f3f1b4d9bf682c4f1b5efb55b9714dc1bcc9eb61cf514fce28a5ed1950652cf9506cabd59b9ae9399bdd03d0c3b25d4e459fe07685d3a569979ea5c35ef30f442a447873821b313588be9c58a19f8b1180be5cd1266486768adc8c6059f1d46df8da678b871f7c014d205acd6e316160a516bc0f1d35b82a47670a58fa49444501277c2fc30d9b19dfacd59c8641272602260096ba8e0f8137346429db0040a18354e53f1b5ce6bbcd511ce901170aa46672823434e41afe4855fd83a7a7a60fe92bb5597ff3715ace02a71b6f10ff532b0df8b7e0ab9886b2f317942f70d320f8029-9f3a9c4c20b54931c201618c311e6e8e412002fa2dc4932883a5b44",
    "result": [
        {
            "_ref": "network/ZG5zLm5ldHdvcmskMTAuMTQ4LjEyLjAvMjIvMA:127.1.0.0/16/default",
            "comment": "Some comment",
            "network": "127.1.0.0/16",
            "network_view": "default"
        },
        {
            "_ref": "network/ZG5zLm5ldHdvcmskMTAuMTQ4LjEyNC4wLzIyLzA:127.2.0.0/16/default",
            "comment": "Some comment here as well",
            "network": "127.2.0.0/16",
            "network_view": "default"
        }
    ]
}
```
### Non-paging Request
`curl -H "Content-Type: application/json" -k1 -u admin -X GET  'https://ib-prod.example.com/wapi/v2.5/network?_max_results=2&_proxy_search=GM'`

### Non-paging Response
```json
[
    {
        "_ref": "network/ZG5zLm5ldHdvcmskMTAuMTQ4LjEyLjAvMjIvMA:127.1.0.0/16/default",
        "comment": "Some comment",
        "network": "127.1.0.0/16",
        "network_view": "default"
    },
    {
        "_ref": "network/ZG5zLm5ldHdvcmskMTAuMTQ4LjEyNC4wLzIyLzA:127.2.0.0/16/default",
        "comment": "Some comment here as well",
        "network": "127.2.0.0/16",
        "network_view": "default"
    }
]
```
For our use case, pagination is a crucial functionality as Infoblox is a critical system.I am wondering if creating a separate Find function that returns (*Output, error) would be a better approach.
Thanks for your quick response, I appreciate it. :)

Reply:
I just pushed a new function called Query that has three parameters - conditions, options and out ( type interface{} ). With the out parameter users can define and parse whatever objects they want without the need to modify the library. :) Thanks